File: flowcore/src/main/java/com/raizlabs/android/dbflow/annotation/Unique.java
package com.raizlabs.android.dbflow.annotation;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * Description: Marks the field as unique, meaning its value cannot be repeated. It is, however,
 * NOT a primary key.
 */
@Retention(RetentionPolicy.SOURCE)
@Target(ElementType.FIELD)
public @interface Unique {

    /**
     * @return if field is unique. If false, we expect {@link #uniqueGroups()} to be specified.`
     */
    boolean unique() default true;

    /**
     * @return Marks a unique field as part of a unique group. For every unique number entered,
     * it will generate a UNIQUE() column statement.
     */
    int[] uniqueGroups() default {};
}

File: core/src/main/kotlin/com/raizlabs/dbflow5/annotation/Unique.kt
package com.raizlabs.dbflow5.annotation

/**
 * Description: Marks the field as unique, meaning its value cannot be repeated. It is, however,
 * NOT a primary key.
 */
@Retention(AnnotationRetention.BINARY)
@Target(AnnotationTarget.FIELD)
annotation class Unique(
    /**
     * @return if field is unique. If false, we expect [.uniqueGroups] to be specified.`
     */
    val unique: Boolean = true,
    /**
     * @return Marks a unique field as part of a unique group. For every unique number entered,
     * it will generate a UNIQUE() column statement.
     */
    val uniqueGroups: IntArray = [],
    /**
     * Defines how to handle conflicts for a unique column
     *
     * @return a [ConflictAction] enum
     */
    val onUniqueConflict: ConflictAction = ConflictAction.FAIL)
